solidity/test/XConversions.js

```diff
@@ -175,87 +152,12 @@ contract("XConversions", accounts => {
             await reportAndRelease(accounts[5], amount, txId1, EOS_BLOCKCHAIN, xTransferId1)
             await reportAndRelease(accounts[4], amount, txId2, EOS_BLOCKCHAIN, xTransferId2)
 
-            await utils.catchRevert(bntConverter.completeXConversion(
-                path,                                              
-                1,                                               
-                xTransferId2,                                       
-                0,                                                    
-                0,                                                      
-                utils.zeroBytes32,                                                      
-                utils.zeroBytes32,                                                      
-                { from: accounts[5] }
-            ))
-        })
-
-        it("should be able to completeXConversion2 to eth", async () => {
-            const txId = getId()
-            const xTransferId = getId()
-            const amount = web3.toWei('10') // releasing 10 BNT
-            const path = bntEthPath
-
-            await reportAndRelease(accounts[5], amount, txId, EOS_BLOCKCHAIN, xTransferId)
-
-            const prevBalance = await web3.eth.getBalance(accounts[5])
-
-            const res = await bntConverter.completeXConversion2(
-                path,
-                1,
-                xTransferId,
-                { from: accounts[5] }
-            )
-
-            const currBalance = await web3.eth.getBalance(accounts[5])
-
-            assert(currBalance.greaterThan(prevBalance))
-        })
-
-        it("should be able to completeXConversion2 to an ERC20", async () => {
-            const txId = getId()
-            const xTransferId = getId()
-            const amount = web3.toWei('10') // releasing 10 BNT
-            const path = bntErc20Path
-
-            await reportAndRelease(accounts[5], amount, txId, EOS_BLOCKCHAIN, xTransferId)
-
-            const prevBalance = await erc20Token.balanceOf(accounts[5])
-
-            const retAmount = await bntConverter.completeXConversion2.call(
-                path,                                              
-                1,                                               
-                xTransferId,                                       
-                { from: accounts[5] }
-            )
-
-            const res = await bntConverter.completeXConversion2(
-                path,                                              
-                1,                                               
-                xTransferId,                                       
-                { from: accounts[5] }
-            )
-
-            const currBalance = await erc20Token.balanceOf(accounts[5])
+            await bntToken.approve(bancorNetwork.address, 0, { from: accounts[5] })
```

Comment:
@yudilevi : This line implies a stateful test. How come the allowance is larger than 0 (to require executing this "0-approve" transaction)? We should strive for stateless tests ("it"s).